prototypes/brownian_01.ipynb

```diff
@@ -0,0 +1,925 @@
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "statearray = StateArray.from_states(states)\n",
+    "a = statearray[:4].reshape(2, 2)\n",
+    "a"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "63c41b5d-5cb5-4739-88a5-a0bec95fff43",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<State epoch=2022-01-04 00:00:00.000 x=7.215816384842416 km y=2.41404569941827 km>"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "a[1, 1]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "f94b4680-1c56-4463-a681-8707c8f568d9",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "(<Quantity [8.38625953, 9.62613917, 4.70111096, 7.60891732, 2.76699349] km>,\n",
+       " <Quantity [1.0229781 , 1.31156493, 1.0908986 , 0.32284445, 0.90151771] rad>)"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "statearray.to_polar()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "242c8234-446f-405f-81d7-4965338f599f",
+   "metadata": {},
+   "source": [
+    "Just like the `Orbit` class is a wrapper around the `State` class in many ways, the `Orbit` array mostly wraps `StateArray`. Its `propagate` method could theoretically allow standard `numpy` broadcasting rules depending on shape of the `timedelta` parameter. In the following example, for simplicity, only scalar values for `timedelta` or values for `timedelta` with a shape matching the shape of the `OrbitArray` are allowed:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "33a90d57-06f3-4ef3-8e26-21e3189fa601",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@typechecked\n",
+    "class OrbitArray:\n",
+    "    \n",
+    "    def __init__(self, statearray: StateArray):\n",
+    "        self._statearray = statearray\n",
+    "    \n",
+    "    def __repr__(self) -> str:\n",
+    "        return  (\n",
+    "            f'<OrbitArray shape={self.shape} value=[\\n'\n",
+    "            + '\\n'.join([\n",
+    "                f' (epoch={state.epoch} x={state.x} y={state.y}),'\n",
+    "                for state in self._statearray.reshape(np.multiply.reduce(self._statearray.shape))\n",
+    "            ])\n",
+    "            + '\\n]>'\n",
+    "        )\n",
+    "    \n",
+    "    def __getitem__(self, idx) -> 'Union[OrbitArray, Orbit]':\n",
+    "        target = self._statearray[idx]\n",
+    "        if isinstance(target, State):\n",
+    "            return Orbit(state = target)\n",
+    "        return OrbitArray(statearray = target)\n",
+    "\n",
+    "    def propagate(self, timedelta: TimeDelta, inplace: bool = False) -> 'OrbitArray':\n",
```

Comment:
Same comment as above, let's not close the door on `inplace=True` but let's not do this in this first iteration so we can focus.

But @s-m-e is *veeeeeeeeeeeeeeeeeeeery* stubborn and wants to burn hours writing 2x as many tests 😉 so let's give him the chance!